src/game/memory.h

```diff
@@ -48,16 +40,113 @@ void *segmented_to_virtual(const void *addr);
 void *virtual_to_segmented(u32 segment, const void *addr);
 void move_segment_table_to_dmem(void);
 
-void main_pool_init(void *start, void *end);
-void *main_pool_alloc(u32 size, u32 side);
-u32 main_pool_free(void *addr);
-void *main_pool_realloc(void *addr, u32 size);
+struct MainPoolRegion {
+    u8* start;
+    u8* end;
+};
+
+#define MAIN_POOL_REGIONS_COUNT 2
```

Comment:
If regions are still specifiable and involve more than one use in HackerSM64, then we should still be using enums to specify region IDs.

Reply:
Reworked to enums.